src/Microsoft.TestPlatform.CrossPlatEngine/Discovery/DiscoveryContext.cs

```diff
@@ -14,5 +23,44 @@ public class DiscoveryContext : IDiscoveryContext
         /// Gets the run settings specified for this request.
         /// </summary>
         public IRunSettings RunSettings { get; internal set; }
+
+        /// <summary>
+        /// Returns TestCaseFilterExpression validated for supportedProperties.
+        /// If there is a parsing error in filter expression, TestPlatformFormatException() is thrown.
+        /// </summary>
+        /// <param name="supportedProperties"> The supported Properties. </param>
+        /// <param name="propertyProvider"> The property Provider. </param>
+        /// <returns> The <see cref="ITestCaseFilterExpression"/>. </returns>
+        public virtual ITestCaseFilterExpression GetTestCaseFilter(IEnumerable<string> supportedProperties, Func<string, TestProperty> propertyProvider)
+        {
+            TestCaseFilterExpression adapterSpecificTestCaseFilter = null;
+            if (this.FilterExpressionWrapper != null)
+            {
+                if (!string.IsNullOrEmpty(this.FilterExpressionWrapper.ParseError))
+                {
+                    throw new TestPlatformFormatException(this.FilterExpressionWrapper.ParseError, this.FilterExpressionWrapper.FilterString);
+                }
+
+                adapterSpecificTestCaseFilter = new TestCaseFilterExpression(this.FilterExpressionWrapper);
+                var invalidProperties = adapterSpecificTestCaseFilter.ValidForProperties(supportedProperties, propertyProvider);
+
+                if (invalidProperties != null)
+                {
+                    var invalidPropertiesString = string.Join(CrossPlatEngineResources.StringSeperator, invalidProperties);
+                    var validPropertiesSttring = supportedProperties == null ? string.Empty : string.Join(CrossPlatEngineResources.StringSeperator, supportedProperties.ToArray());
+                    var errorMessage = string.Format(CultureInfo.CurrentCulture, CrossPlatEngineResources.UnsupportedPropertiesInTestCaseFilter, invalidPropertiesString, validPropertiesSttring);
+
+                    // For unsupported property don�t throw exception, just log the message. Later it is going to handle properly with TestCaseFilterExpression.MatchTestCase().
+                    EqtTrace.Info(errorMessage);
+                }
+            }
+
+            return adapterSpecificTestCaseFilter;
+        }
+
+        /// <summary>
+        /// Gets or sets the FilterExpressionWrapper instance as created from filter string.
+        /// </summary>
+        internal FilterExpressionWrapper FilterExpressionWrapper {get; set; }
```

Comment:
private?

Reply:
FilterExpressionWrapper is set by DiscovererEnumerator in case of discovery and by RunTestsWithSources in case of test run. Thus cant be private.